src/Service/ProcessServiceImpl.cpp

```diff
@@ -82,8 +83,10 @@ Status ProcessServiceImpl::GetProcessMemory(ServerContext*, const GetProcessMemo
   uint64_t size = std::min(request->size(), kMaxGetProcessMemoryResponseSize);
   response->mutable_memory()->resize(size);
   uint64_t num_bytes_read = 0;
-  if (utils::ReadProcessMemory(request->pid(), request->address(),
-                               response->mutable_memory()->data(), size, &num_bytes_read)) {
+  uint32_t pid = request->pid();
```

Comment:
Can you just do ToNativeProcessId(request->pid()) here? It already has a check right?

Reply:
Well ReadProcessMemory will be implemented for Windows too, we might as well use the platform agnostic pid. I removed the check here and added `ToNativeProcessId` in the implementation of ReadProcessMemory.